pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuStateVecError.hpp

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022-2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Adapted from JET: https://github.com/XanaduAI/jet.git
+// and from Lightning: https://github.com/PennylaneAI/pennylane-lightning.git
+
+/**
+ * @file cuStateVecError.hpp
+ * Defines macros that throws Exception from cuStateVec failure error codes.
+ */
+
+#pragma once
+#include <string>
```

Comment:
You may want to use `string_view` in GetCuStateVecErrorString.
```suggestion
#include <string_view>
#include <string>
```